src/Talis/Persona/Client/OAuthClients.php

```diff
@@ -0,0 +1,154 @@
+<?php
+namespace Talis\Persona\Client;
+
+class OAuthClients extends Base
+{
+    /**
+     * Return an outh client
+     * @param string $clientId
+     * @param string $token
+     * @return array
+     * @throws \Exception
+     */
+    public function getOAuthClient($clientId, $token)
+    {
+        if(!is_string($clientId) || trim($clientId) === '')
+        {
+            throw new \InvalidArgumentException("Invalid clientId");
+        }
+        if(!is_string($token) || trim($token) === '')
+        {
+            throw new \InvalidArgumentException("Invalid token");
+        }
+
+        $url = $this->config['persona_host'].'/clients/'.$clientId;
+
+        try
+        {
+            $client = $this->personaGetOAuthClient($url, $token);
+            return $client;
+        } catch(\Exception $e)
+        {
+            throw new \Exception('OAuth client not found');
```

Comment:
I've raised an issue for this: https://github.com/talis/persona-php-client/issues/10